src/app/GitUI/UserControls/FileStatusList.cs

```diff
@@ -313,39 +339,31 @@ public override ContextMenuStrip ContextMenuStrip
         public Func<ObjectId?, string>? DescribeRevision { get; set; }
 
         public bool FilterFocused => FilterComboBox.Focused;
-        public bool SearchFocused => SearchComboBox.Focused;
+        public bool SearchFocused => GitGrepSearchComboBox.Focused;
 
         /// <summary>
-        /// Enable Search Commit combobox for the file list.
-        /// Refresh of list must be called explicitly.
+        ///  Indicates whether the git-grep search functionality is enabled for this control.
         /// </summary>
-        public bool SearchEnabledForList
-        {
-            private get => _searchEnabledForList;
-            set
-            {
-                _searchEnabledForList = value;
-                EnableSearchForList(value && AppSettings.ShowSearchCommit.Value);
-            }
-        }
+        [DefaultValue(false)]
+        public bool CanUseGitGrepSearch { get; set; }
 
-        private void EnableSearchForList(bool enable)
+        public void ToggleGitGrepSearch(bool visible)
```

Comment:
Toggle is not a good name, it updates the status at every update.
I have some doubts for some of the changes, not time to investigate today....

Reply:
> it updates the status at every update.

Which "status" are you talking about?